src/map/GMap.js

```diff
@@ -238,6 +239,25 @@
         this._prevCenterLat = this.centerLat();
         this._prevCenterLong = this.centerLong();
         this._prevZoom = this.zoom();
+        
+        if (this.mapDrawingTools()) {
```

Comment:
Moving the initialization of the drawing tools to the update function makes sense, I'll see to that.  I'll also add a way to keep track of and get/set the drawn objects.